public/src/pages/TeacherPage.tsx

```diff
@@ -348,32 +350,25 @@ export class TeacherPage extends ViewPage {
         }
     }
 
-    private async getCourses(): Promise<Course[]> {
+    private async getCourses(statuses: Enrollment.UserStatus[]): Promise<Course[]> {
         const curUsr = this.userMan.getCurrentUser();
         if (curUsr) {
-            return this.courseMan.getCoursesForUser(curUsr, []);
+            const courses = await this.courseMan.getCoursesForUser(curUsr, statuses);
+            const enrols = await this.courseMan.getEnrollmentsForUser(curUsr.getId());
```

Comment:
Looks like there are two round-trips to the server here... can’t we avoid that

Reply:
Sure, I can teach the `getCoursesForUser` method to take the visibility state into consideration. But note that this way we move all the sorting from the client to the server. Is it ok?
I suppose it is because it's still better than two consecutive server calls.

Reply:
I prefer to do more on the server, where that makes sense of course. But I also don't want to introduce yet another RPC method (adding to the interface) just to support more functionality when it can be accomplished with existing methods, albeit with some minor added complexity in the frontend. Changing the function to support more needs is ok, unless it becomes "artificial"...

Reply:
Fixed that.